apps/backend/src/authn/oidc.strategy.ts

```diff
@@ -0,0 +1,42 @@
+import {Injectable, UnauthorizedException} from '@nestjs/common';
+import {PassportStrategy} from '@nestjs/passport';
+import {Strategy} from 'passport-openidconnect';
+import {ConfigService} from '../config/config.service';
+import {AuthnService} from './authn.service';
+
+@Injectable()
+export class OidcStrategy extends PassportStrategy(Strategy, 'oidc') {
+  constructor(
+    private readonly authnService: AuthnService,
+    private readonly configService: ConfigService
+  ) {
+    super(
+      {
+        issuer: configService.get('OIDC_ISSUER') || 'disabled',
+        authorizationURL:
+          configService.get('OIDC_AUTHORIZATION_URL') || 'disabled',
+        tokenURL: configService.get('OIDC_TOKEN_URL') || 'disabled',
+        userInfoURL: configService.get('OIDC_USER_INFO_URL') || 'disabled',
+        clientID: configService.get('OIDC_CLIENTID') || 'disabled',
+        clientSecret: configService.get('OIDC_CLIENT_SECRET') || 'disabled',
+        callbackURL: `${configService.get('EXTERNAL_URL')}/authn/oidc/callback`,
```

Comment:
This may require some documentation for users to know what they need to set their callback URL to on their OpenID provider.

Reply:
Mentioned as part of #889